mantidproject/mantid
# Remove Notebook and mantidpython

Separately the [PythonNotebook docs page](https://docs.mantidproject.org/v6.1.0/concepts/PythonNotebook.html) has been updated on main (not on the website yet) [in this PR](https://github.com/mantidproject/mantid/pull/34450).

**Description of work.**
- Remove the mantidpython and mantidnotebook. They were both separate launcher scripts, and the mantidpython was also used for running our systemtests and python unittests. Now, to access the mantid api, the PYTHONPATH is set to the build/bin directory to run the tests. With an installed package of mantidworkbench, you can just launch the bundled python as detailed here: https://github.com/mantidproject/mantid/pull/34450/files#diff-dab7a662c04b0c0ab7ee1ec2aa593b9d4a8830b2a2659f08ee43e550047adf40R26-R28 Users are expected to install the notebook python module themselves.
- Removed the .deb and .rpm package scripts
- Removed all references to ``mantidpython`` in code and docs (apart from old release notes)
- Removed all references to ``notebook`` in code (apart from algorithms that write notebooks such as NotebookBuilder, NotebookWriter, GenerateIPythonNotebook, Generate Notebook, DataProcessorWidget)
and docs (apart from old release notes and a few parts of the dev-docs that I think are fine!) 
**To test:**
1) Jenkins
- PR builds check that unit tests ran
- Run SEPARATE build_package_with_branch jobs (must be separate for systemtests to run) for Windows and MacOS to check the systemtests run ( I got a recent version of this PR to pass on build numbers 78 and 76 respectively).

2) Check Jemalloc is used for tests on Linux
- On Linux, run the test-pr command to get this branch, activate the dev conda environment and build
- Check jemalloc is used for unittests with this command giving jemalloc in it's output: ``ctest -V -R AnalysisData -N``
- start the system tests running with `.systemtest` in the build folder. Wait til they are actually running and not downloading data. 
- Then in another terminal run `ps ax | grep runSystemTests` and note the process ID
- Now run `cat /proc/PID/smaps | grep jemalloc` but replace PID with the process ID from the last step and you should get some output which confirms jemalloc is being used by the systemtests!

3) check the docs updates are okay (a code review should suffice).
4) Finally some git grepping!
- In the source directory run `git grep -n "mantidpython"` and check the list is as described in the "Description of work" above.
- Run `git grep -n "notebook"` and check the list is as described in the "Description of work" above. You may wish to cd into docs or dev-docs to make the list more digest-able.

Fixes #34388

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
